File: src/main/scala/awsprog/Grid.scala
package awsprog

import java.awt.Color
import java.awt.Graphics
import java.awt.Point

import javax.swing.JPanel

class Grid(ymax: Int, xmax: Int) extends JPanel {
  val interval: Int = 10
  val xgridMax = xmax * interval
  val ygridMax = ymax * interval
  var cells: List[Point] = List()
  
  override def paintComponent(graphic: Graphics): Unit = {
    super.paintComponent(graphic)
    cells.foreach { cell => 
     val cellX = interval + (cell.x * interval)
     val cellY = interval + (cell.y * interval)
     graphic.setColor(Color.RED)
     graphic.fillRect(cellX,cellY, interval, interval)
    }
    graphic.setColor(Color.BLACK)
    graphic.drawRect(interval, interval, ygridMax, xgridMax)
    
    (interval until (ygridMax + interval) by interval)
      .foreach(y => graphic.drawLine(y, interval, y, xgridMax + interval))
      
    (interval until (xgridMax + interval) by interval)
      .foreach(x => graphic.drawLine(interval, x, ygridMax + interval, x))
  }
  
  def clear(): Unit =  cells = List()

  def fillCell(x: Int, y: Int): Unit =  cells = cells.+:(new Point(x,y))
}

File: src/main/java/awsprog/Grid.java
package awsprog;
import java.awt.Color;
import java.awt.Graphics;
import java.awt.Point;
import java.util.ArrayList;
import java.util.List;

import javax.swing.JPanel;


public class Grid extends JPanel {

  /**
   * 
   */
  private static final long serialVersionUID = 3155541099893860655L;
  private List<Point> fillCells;
  private final int ygridmax;
  private final int xgridmax;
  private final int interval = 10;

  public Grid(int ymax, int xmax) {
    this.ygridmax = ymax * interval;
    this.xgridmax = xmax * interval;
    fillCells = new ArrayList<Point>(25);
  }

  @Override
  protected void paintComponent(Graphics g) {
      super.paintComponent(g);
      for (Point fillCell : fillCells) {
          int cellX = interval + (fillCell.x * interval);
          int cellY = interval + (fillCell.y * interval);
          g.setColor(Color.RED);
          g.fillRect(cellX, cellY, interval, interval);
      }
      g.setColor(Color.BLACK);
      g.drawRect(interval, interval, ygridmax, xgridmax);

      for (int i = interval; i <= ygridmax; i += interval) {
          g.drawLine(i, interval, i, xgridmax + interval);
      }

      for (int i = interval; i <= xgridmax; i += interval) {
          g.drawLine(interval, i, ygridmax + interval, i);
      }
  }
  
  public void clear() {
    fillCells.clear();
  }

  public void fillCell(int x, int y) {
      fillCells.add(new Point(x, y));
  }
}
